elyra/pipeline/processor_kfp.py

```diff
@@ -480,26 +486,48 @@ def _cc_pipeline(self,
                                                           cos_directory,
                                                           operation)
 
-            # If operation is a "non-standard" component, load it's spec and create operation with factory function
+            # If operation is a "non-standard" component_id, load it's spec and create operation with factory function
             else:
                 component_source = {}
                 component_source[operation.component_source_type] = operation.component_source
 
-                # Build component task factory
+                # Change value of variables according to their type. Path variables should include
+                # the contents of the specified file and dictionary values must be converted from strings.
+                component = self._component_registry.get_component(operation.classifier)
+                for component_property in component.properties:
+                    if component_property.ref in ['runtime_image', 'component_source', 'component_source_type']:
```

Comment:
See earlier comment.

Reply:
we will refactor the generic components in a future pr